pkg/experiment/metastore/metastore_readindex.go

```diff
@@ -5,128 +5,65 @@ import (
 	"fmt"
 	"time"
 
-	"github.com/go-kit/log"
-	"github.com/google/uuid"
-
 	metastorev1 "github.com/grafana/pyroscope/api/gen/proto/go/metastore/v1"
 )
 
-var tcheckFreq = 10 * time.Millisecond
-
-func (m *Metastore) ReadIndex(ctx context.Context, req *metastorev1.ReadIndexRequest) (*metastorev1.ReadIndexResponse, error) {
-	//todo
-	//If the leader has not yet marked an entry from its current term committed, it waits until it
-	//has done so. The Leader Completeness Property guarantees that a leader has all committed
-	//entries, but at the start of its term, it may not know which those are. To find out, it needs to
-	//commit an entry from its term. Raft handles this by having each leader commit a blank no-op
-	//entry into the log at the start of its term. As soon as this no-op entry is committed, the leader’s
-	//commit index will be at least as large as any other servers’ during its term.
-	t := time.Now()
-	readIndex := m.raft.CommitIndex()
-	raftLogger := func() log.Logger {
-		return log.With(m.logger, "component", "raft_debug",
-			"request_id", req.DebugRequestId,
-			"op", "ReadIndex",
-			"read_index", readIndex,
-			"applied_index", m.raft.AppliedIndex(),
-			"commit_index", m.raft.CommitIndex(),
-			"last_index", m.raft.LastIndex(),
-			"duration", time.Since(t),
-		)
-	}
+const tCheckFreq = 10 * time.Millisecond
 
-	raftLogger().Log("msg", "verify_leader")
+// ReadIndex returns the current commit index and verifies leadership.
+func (m *Metastore) ReadIndex(_ context.Context, req *metastorev1.ReadIndexRequest) (*metastorev1.ReadIndexResponse, error) {
+	commitIndex := m.raft.CommitIndex()
 	if err := m.raft.VerifyLeader().Error(); err != nil {
 		return nil, wrapRetryableErrorWithRaftDetails(err, m.raft)
 	}
+	return &metastorev1.ReadIndexResponse{ReadIndex: commitIndex}, nil
+}
+
+// readIndex ensures the node is up-to-date for read operations,
+// providing linearizable read semantics. It calls the ReadIndex RPC
+// and waits for the local applied index to catch up to the returned read index.
+// This method should be used before performing local reads to ensure consistency.
+func (m *Metastore) readIndex(ctx context.Context) error {
```

Comment:
I may sound silly, but I propose naming this function in an absurdly verbose way:
```go
func (m *Metastore) waitLeaderCommitIndexAppliedLocally(context.Context) error
```